vsintegration/src/FSharp.Editor/Common/CommonHelpers.fs

```diff
@@ -411,25 +411,31 @@ module internal Extensions =
                 return! this.ParseDocument(document, options, sourceText.ToString())
             }
 
-        member this.ParseAndCheckDocument(filePath: string, textVersionHash: int, sourceText: string, options: FSharpProjectOptions) : Async<(Ast.ParsedInput * FSharpCheckFileResults) option> =
+        member this.ParseAndCheckDocument(filePath: string, textVersionHash: int, sourceText: string, options: FSharpProjectOptions, ?allowStaleResults: bool) : Async<(FSharpParseFileResults * Ast.ParsedInput * FSharpCheckFileResults) option> =
             async {
-                let! parseResults, checkFileAnswer = this.ParseAndCheckFileInProject(filePath, textVersionHash, sourceText, options)
-                return
+                match defaultArg allowStaleResults true, this.TryGetRecentCheckResultsForFile(filePath, options) with
```

Comment:
I thought the plan was to try to get correct, non-stale results under a timeout/cancellation?  Is that feasible to prototype?

I'm very loathe to return to a situation where we rely on stale results even in very small files and scripts...